modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/utils/fetchUtil.ts

```diff
@@ -58,26 +58,13 @@ export async function getAllItems<T>({
 	filter?: string;
 	url: string;
 }) {
-	let allItems: T[] = [];
-	let currentPage = 1;
-	let lastPage;
+	const {items} = await fetchJSON<{items: T[]}>({
+		input: filter
+			? `${url}?filter=${filter}&page=${'-1'}`
```

Comment:
I think that you can concatenate the string like: from `page=${'-1'}` to `page=-1`. What do you think?